src/components/application_manager/test/resumption/resume_ctrl_test.cc

```diff
@@ -567,80 +589,137 @@ TEST_F(ResumeCtrlTest, SetupDefaultHMILevel) {
   res_ctrl->SetupDefaultHMILevel(app_mock);
 }
 
+TEST_F(ResumeCtrlTest, ApplicationResumptiOnTimer_AppInFull) {
+  EXPECT_CALL(mock_app_mngr_, application(_)).WillRepeatedly(Return(app_mock));
+
+  mobile_apis::HMILevel::eType restored_test_type = eType::HMI_FULL;
+  const uint32_t ign_off_count = 0;
+  smart_objects::SmartObject saved_app;
+  saved_app[application_manager::strings::ign_off_count] = ign_off_count;
+  saved_app[application_manager::strings::hmi_level] = restored_test_type;
+
+  MockStateController state_controller_;
+  EXPECT_CALL(mock_app_mngr_, state_controller())
+      .WillOnce(ReturnRef(state_controller_));
+  EXPECT_CALL(state_controller_, SetRegularState(_, restored_test_type))
+      .Times(AtLeast(1));
+  GetInfoFromApp();
+  EXPECT_CALL(*mock_storage, GetSavedApplication(kTestPolicyAppId_, _, _))
+      .Times(2)
+      .WillRepeatedly(DoAll(SetArgReferee<2>(saved_app), Return(true)));
+
+  EXPECT_CALL(*mock_storage, RemoveApplicationFromSaved(_, _))
+      .WillOnce(Return(true));
+
+  EXPECT_CALL(mock_app_mngr_, GetUserConsentForDevice("12345"))
```

Comment:
@Kozoriz 
Looks like this device id is set in some other place, so maybe it worth to use variable instead?